src/simulate.py

```diff
@@ -341,9 +349,46 @@ def simulate(self, computation_dfg, hw):
                         * hw.latency[node_data["function"]] # ns
                     )
 
+                # wires
+                active_net_len = len(active_net)
+                computation_nodes = list(computation_dfg.out_edges(node))
+                node_name = node_data["allocation"]
+                node_bool = "allocation" in node_data and node_name != "" and "Mem" not in node_name and "Buf" not in node_name
+               
+                if node_bool:
+                    for comp_node in computation_nodes:
+                        child_node_data = computation_dfg.nodes[comp_node[1]]
+                        child_node_name = child_node_data["allocation"]
+                        comp_node_bool = "allocation" in child_node_data and child_node_name != "" and "Mem" not in child_node_name and "Buf" not in child_node_name
+                        if comp_node_bool:
+
+                            # temporary solution, will be changed when computation_dfg gets fixed 
+                            parasitic_edge = None
+                            if not hw.parasitic_graph.has_edge(node_name, child_node_name): 
+                                parasitic_nodes_out = list(hw.parasitic_graph.out_edges(node_name))
+                                child_node_function = child_node_data["function"]
+                                for node in parasitic_nodes_out:
+                                    if child_node_function in node[1]:
+                                        child_node_name = node[1]
+                                        break
+                                parasitic_edge = hw.parasitic_graph[node_name][child_node_name]
+                            else : 
+                                parasitic_edge = hw.parasitic_graph[node_name][child_node_name]
+                            active_net.add("({}, {})".format(node_name, child_node_name))
+                            # print("({}, {})".format(node_name, comp_node_data["allocation"]))
+
+                            if active_net_len != len(active_net):
+                                active_cap_vals.append(parasitic_edge["net_cap"])
```

Comment:
can you wrap all of this in a separate function. make the inputs and outputs explicit though as much as possible. instance variables don't need to be passed explicitly, but comptation_nodes (rename out_nodes) should be.